posthog/queries/trends/sql.py

```diff
@@ -395,13 +395,13 @@
     arraySort(groupUniqArray(dateTrunc(%(interval)s, toTimeZone(toDateTime(events.timestamp, %(timezone)s), %(timezone)s)))) AS all_activity,
     arrayPopBack(arrayPushFront(all_activity, dateTrunc(%(interval)s, toTimeZone(toDateTime(min({created_at_clause}), %(timezone)s), %(timezone)s)))) as previous_activity,
     arrayPopFront(arrayPushBack(all_activity, dateTrunc(%(interval)s, toDateTime('1970-01-01')))) as following_activity,
-    arrayMap((previous,current) -> if(
+    arrayMap((previous,current, index) -> if(
         previous = current, 'new', if(
-                current - INTERVAL 1 {interval} = previous,
+                current - INTERVAL 1 {interval} = previous AND index != 1,
```

Comment:
Hmm, not 100% sure of this logic, but the idea is:

current = all events seen
previous = [created_at, ...].

At the first index, what we have is the first event, so it can never be 'returning', only new or resurrecting. And new is already handled.